pkg/kubelet/cm/topologymanager/scope_container.go

```diff
@@ -55,13 +55,15 @@ func (s *containerScope) Admit(pod *v1.Pod) lifecycle.PodAdmitResult {
 		if !admit {
 			return topologyAffinityError()
 		}
-
+		s.Lock()
 		if (s.podTopologyHints)[string(pod.UID)] == nil {
 			(s.podTopologyHints)[string(pod.UID)] = make(map[string]TopologyHint)
 		}
 
 		klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
 		(s.podTopologyHints)[string(pod.UID)][container.Name] = bestHint
+		s.Unlock()
+
```

Comment:
And this would become:
```
klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
s.setTopologyHints(string(pod.UID), container.Name, bestHint)
```